cli/src/cmd/forge/coverage.rs

```diff
@@ -96,11 +104,41 @@ impl CoverageArgs {
         let project = {
             let mut project = config.ephemeral_no_artifacts_project()?;
 
-            // Disable the optimizer for more accurate source maps
-            project.solc_config.settings.optimizer.disable();
-            project.solc_config.settings.optimizer.runs = None;
-            project.solc_config.settings.optimizer.details = None;
-            project.solc_config.settings.via_ir = None;
+            if self.ir_minimum {
+                // TODO: How to detect solc version if the user does not specify a solc version in
+                // config  case1: specify local installed solc ?
+                //  case2: mutliple solc versions used and  auto_detect_solc == true
```

Comment:
`vir-ir` and `ir-minimum` will only work in solidity 0.8.13 and above. 
I don't find a better way to reject the unsatisfied compilers.